table.cpp

```diff
@@ -457,12 +457,21 @@ void table_t::escape_type(const string &value, ColumnType flags)
     switch (flags) {
     case COLUMN_TYPE_INT: {
         // For integers we take the first number, or the average if it's a-b
-        long from, to;
-        int items = sscanf(value.c_str(), "%ld-%ld", &from, &to);
-        if (items == 1) {
+        int64_t from, to;
+        // limit number of digits parsed to avoid undefined behaviour in sscanf
+        int items = sscanf(value.c_str(), "%18ld-%18ld", &from, &to);
```

Comment:
My reading of the sscanf man page is that you need to use '%18Ld' here to really get a 64bit value even on 32bit machines. You also need to use 'long long' then as data type for your variables to guarantee that everything matches. This should fix the Travis fail on OSX.

Reply:
Great, I've taken that approach

Reply:
Unfortunately, that's not supported in C++11:  `table.cpp:462:43: error: ISO C++11 does not support the ‘%Ld’ gnu_scanf format`

Reply:
OK, anyone got any suggestions of what integer type we can use and what format string to pair it with? It needs to parse *more than 32 bits* so that we can constrain it with a digit specifier and still get the full 32 bit output.

So far we've tried 
* `int64_t` with `%18ld` (macosx didn't like this)
* `long long` with `%18Ld` (no L in C++11)

Reply:
The recommendation (in the BUG section of scanf. Who reads that far down? ;) ) is '%18lld' (and `long long`). Can you try that? If it doesn't work either, then we have to go the `strtol` route. I can help out in that case.

Reply:
The CI is green with '%18lld' - thanks!